src/components/ThePostHeader.vue

```diff
@@ -0,0 +1,252 @@
+                    {{ $t('archive') }}
+                  </a>
+                  <router-link v-if="postUserId === userId" class="dropdown-item"
+                    :to="{
+                      name: 'write',
+                      params: {
+                        postId: post.id
+                      }
+                    }">
+                      {{ $t('edit') }}
+                  </router-link>
+                  <a v-if="postUserId === userId"
+                    @click="deletePost"
+                    href="#"
+                    class="dropdown-item">
+                    {{ $t('delete') }}
+                  </a>
+                  <a v-else class="dropdown-item"
+                    @click="report"
+                    :class="{ 'is-loading': isReporting }">
+                    {{ $t('report') }}
+                  </a>
+                </div>
+              </div>
+            </div>
+          </span>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { archivePost, reportPost, deletePost, getAttachmentUrls } from '@/api'
+import { date } from '@/helper.js'
+import TextEditor from '../components/TheTextEditor'
+import PostStatus from '@/components/PostStatus.vue'
+
+export default {
+  name: 'the-post-header',
+  props: {
+    post: { required: true }
+  },
+  data () {
+    return {
+      isArchiving: false,
+      isReporting: false,
+      attachments: null
+    }
+  },
+  computed: {
+    userPictureUrl () {
+      return this.post.created_by.profile.picture
+    },
+    // @TODO: I18n
+    postAuthor () {
+      return this.post.created_by && this.post.created_by.profile.nickname
+    },
+    postCreatedAt () {
+      return date(this.post.created_at)
+    },
+    postUserId () {
+      return this.post.created_by.profile.user_id
+    },
+    ...mapGetters([ 'userId' ])
+  },
+  watch: {
+    'post.attachments': {
+      async handler (attachments) {
+        const results = await getAttachmentUrls(attachments)
+        this.attachments = results.map(({ data }) => ({
+          url: data.file,
+          file: decodeURIComponent(new URL(data.file).pathname.split('/').pop()),
+          id: data.id
+        }))
+      },
+      immediate: true
+    }
+  },
+  methods: {
+    async archive () {
+      this.isArchiving = true
+      await archivePost(this.post.id)
+      this.isArchiving = false
+    },
+    async report () {
+      this.isReporting = true
+      await reportPost(this.post.id)
+      this.isReporting = false
+    },
+    async deletePost () {
+      await deletePost(this.post.id)
+      this.$router.push('/')
+    }
+  },
+  components: {
+    TextEditor, PostStatus
+  }
+}
+</script>
+
+<i18n>
+ko:
+  archive: '담아두기'
+  edit: '수정'
+  delete: '삭제'
+  report: '신고'
+en:
+  archive: 'Archive'
+  edit: 'Edit'
+  delete: 'Delete'
+  report: 'Report'
+</i18n>
+
+<style lang="scss" scoped>
+.title {
+  color:#4a4a4a;
```

Comment:
Sorry for not notifying about this, but we have been decided to use CSS variables to describe colors.
For example, instead of using `#a9a9a9`, we have been decided to use `var(--grey-400)`. For more information, please refer to [this diff](https://github.com/sparcs-kaist/new-ara-web/pull/119/files#diff-bab824f582f28aa5e25d39486e91a3d4).

Also it applies to `#metadata`, and `.postheader__time`.